### PR TITLE
Paginate docker images in ImageVersionPicker on Docker page

### PR DESCRIPTION
**Pull request by dherault**
## Summary

Adds pagination for this component:
image

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: www/src/components/repos/Docker.tsx [hunk 1]
```diff
@@ -107,14 +107,27 @@ function ImageVersionPicker({ image }: any) {
...
+  const { data } = useQuery(DOCKER_IMG_Q, {
+    variables: {
+      dockerRepositoryId: dockerRepository.id,
+      cursor,
+    },
   })
 
-  if (!data || loading) return null
+  useEffect(() => {
+    if (data?.dockerImages) {
```
Comment (michaeljguarino) [inline www/src/components/repos/Docker.tsx:120]: Is this just going to fetch the entire set of images until it stops?  We should only do it when a user scrolls into view for sure, some of these repos have thousands of images

Comment (maciaszczykm) [reply www/src/components/repos/Docker.tsx:120]: I think in designs for this there was a "view more" button.

**Comment by dherault** [comment]
I'm very unhappy with the end result.

Do to the implementation of the new Select, two problems arise:
- The select closes on fetch more click
- `// @ts-nocheck` is needed at the top of the file because `Select` admits only `ReactElement<ListBoxItemBaseProps> | ReactElement<ListBoxItemBaseProps>[]` as children. 

Moreover when clicking on a paginated version, the page reloads and the pagination resets, leaving the version missing from the select.

I say we revert to my previous "all fetch" implementation.

**Comment by maciaszczykm** [comment]
> The select closes on fetch more click

Would `e.stopPropagation()` help with this as temp workaround?

File: www/src/components/repos/Docker.tsx [hunk 2]
```diff
@@ -157,6 +184,17 @@ function ImageVersionPicker({ image }: any) {
             )}
           />
         ))}
+        {hasNextPage ? (
```
Comment (maciaszczykm) [inline www/src/components/repos/Docker.tsx:187]: Would `{hasNextPage && (` work here instead?

Comment (maciaszczykm) [reply www/src/components/repos/Docker.tsx:187]: Actually, this should be moved to `dropdownFooter` prop. See the example from Storybook:

```
dropdownFooter={
  shownLimit < items.length && (
    <ListBoxFooterPlus>View more</ListBoxFooterPlus>
  )
}
```

Comment (maciaszczykm) [reply www/src/components/repos/Docker.tsx:187]: Dropdown closing on click will also stop being a problem when this prop will be used.

Comment (dherault) [reply www/src/components/repos/Docker.tsx:187]: No... Because of problem number two

Comment (maciaszczykm) [reply www/src/components/repos/Docker.tsx:187]: It was my guess as it works like this here: https://pluralsh-design.web.app/?path=/story/select--default (see "Click me!" button).

Comment (dherault) [reply www/src/components/repos/Docker.tsx:187]: OH that's great!

File: www/src/components/repos/Docker.tsx [hunk 3]
```diff
@@ -1,4 +1,7 @@
+// HACK until Select is fixed to admit any ReactNode
+// @ts-nocheck
```
Comment (maciaszczykm) [inline www/src/components/repos/Docker.tsx:2]: Is this needed?

**Review by maciaszczykm — APPROVED** [review]
LGTM, one question added.

**Pull request merged**